Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py

```diff
@@ -360,10 +364,15 @@ def get_update_incident_request_data(client: AzureSentinelClient, args: Dict[str
         severity = demisto.get(fetched_incident_data, 'properties.severity')
     if not status:
         status = demisto.get(fetched_incident_data, 'properties.status')
-    if user_principal_name:
-        owner = {'userPrincipalName': user_principal_name}
-    if assignee_email:
-        owner['email'] = assignee_email
+    if assignee_objectid:
+        owner = {'objectId': assignee_objectid}
+    else:
+        if user_principal_name:
+            owner = {'userPrincipalName': user_principal_name}
+        if assignee_email:
+            owner['email'] = assignee_email
+    if unassign == 'true':
+        owner = {}
```

Comment:
```suggestion
    if unassign == 'true':
        owner = {}
    elif assignee_objectid:
        owner = {'objectId': assignee_objectid}
    else:
        if user_principal_name:
            owner = {'userPrincipalName': user_principal_name}
        if assignee_email:
            owner['email'] = assignee_email
```
This order makes better understanding of the hierarchy of the arguments